src/main/java/martin/task/Task.java

```diff
@@ -0,0 +1,94 @@
+package martin.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public boolean isDone() {
+        return isDone;
+    }
+
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    public void unmarkAsDone() {
+        isDone = false;
+    }
+
+    public boolean contains(String keyword) {
+        return description.contains(keyword);
+    }
+
+    /**
+     * Convert the task to its file format representation.
+     * 
+     * @return String representation of task for file storage.
+     */
+    public String toFileFormat() {
+        return (isDone ? "1" : "0") + " | " + description;
+    }
+
+    /**
+     * Convert a file format string back to a Task.
+     * 
+     * @param fileFormatString String representation from file.
+     * @return A new Task instance.
+     */
+    public static Task fromFileFormat(String fileFormatString) {
+        String[] parts = fileFormatString.split("\\s*\\|\\s*");
+    
+        String type = parts[0];
+    
+        switch (type) {
```

Comment:
If it's only two conditions, consider using if else instead. But I can see the potential of including todos, events and more subtypes of Tasks in the future.